### PR TITLE
Use HTTPS URL's

### PR DESCRIPTION
**Pull request by kurtmckee**
The patch modifies many URL's to use HTTPS. This includes URL's in:

* Documentation
* Translation files
* Code

The following URL's and files were excluded from consideration:

* XML namespace URL's
* Theme files (Javascript, CSS, etc.)

I'm not sure where to put a note about the change in the CHANGES file, though. Do I create a "Documentation" header? And under 1.7.0 or 1.8.0? I'm open to instructions for how to modify the file, or for someone else to make the change later! =)

### CODE REVIEW

File: sphinx/locale/bn/LC_MESSAGES/sphinx.po [hunk 1]
```diff
@@ -11,7 +11,7 @@ msgstr ""
 "POT-Creation-Date: 2017-04-28 15:54+0000\n"
 "PO-Revision-Date: 2017-04-28 15:59+0000\n"
 "Last-Translator: Takayuki SHIMIZUKAWA <shimizukawa@gmail.com>\n"
-"Language-Team: Bengali (http://www.transifex.com/sphinx-doc/sphinx-1/language/bn/)\n"
+"Language-Team: Bengali (https://www.transifex.com/sphinx-doc/sphinx-1/language/bn/)\n"
```
Comment (tk0miya) [inline sphinx/locale/bn/LC_MESSAGES/sphinx.po]: This is not needed. These .po files are generated by `transifex-client`.
Could you revert them all?

**Review by tk0miya — CHANGES_REQUESTED** [review]
LGTM with nits

**Comment by tk0miya** [comment]
The changes for our document are reflected to sphinx-doc.org on the fly.
So no changes entry is needed.

Thanks,

**Comment by kurtmckee** [comment]
I've rebased and force-pushed. The Transifex change commit is removed from the history now.

File: sphinx/locale/bn/LC_MESSAGES/sphinx.po (hunk 1 above)
Comment (kurtmckee) [reply sphinx/locale/bn/LC_MESSAGES/sphinx.po]: Removed the commit from the history and force-pushed. Let me know if anything else is needed. =)

**Review by tk0miya — APPROVED** [review]

**Comment by tk0miya** [comment]
Thank you for quick response.
Merged.

Thanks!

**Comment by jfbu** [comment]
@tk0miya could this have been merged to 1.7-release? (oh sorry, I see that in practice users can always look at master branch on sphinx-doc.org)

**Pull request merged**